task.sql

```diff
@@ -3,31 +1,55 @@
 CREATE DATABASE ShopDB;
 USE ShopDB;
 
 CREATE TABLE Countries (
-    ID INT,
-    Name VARCHAR(50),
-    PRIMARY KEY (ID)
+  ID INT,
+  Name VARCHAR(50),
+  
+  PRIMARY KEY (ID)
+);
+
+CREATE TABLE Warehouses (
+  ID INT,
+  Name VARCHAR(50),
+  Address VARCHAR(50),
+  CountryID INT,
+
+  PRIMARY KEY (ID),
+  FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION
+);
+
+CREATE TABLE Products (
+  ID INT,
+  Name VARCHAR(50),
+
+  PRIMARY KEY (ID)
 );
 
 CREATE TABLE ProductInventory (
-    ID INT,
-    ProductName VARCHAR(50),
-    WarehouseAmount INT,
-    WarehouseName VARCHAR(50),
-    WarehouseAddress VARCHAR(50), 
-    CountryID INT,
-	FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
-    PRIMARY KEY (ID)
+  ID INT,
+  ProductID INT,
+  Amount INT,
+  WarehouseID INT,
+
+  PRIMARY KEY (ID),
+  FOREIGN KEY (ProductID) REFERENCES Products(ID) ON DELETE NO ACTION,
```

Comment:
The `ON DELETE NO ACTION` constraint means that you cannot delete a product if there are inventory records associated with it. Ensure this is the desired behavior. If you want to allow deletion and automatically remove associated inventory records, consider using `ON DELETE CASCADE`.

Reply:
Fixed